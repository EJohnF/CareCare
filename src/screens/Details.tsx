import React, {FC, useEffect} from 'react';
import {ActivityIndicator, View} from 'react-native';
import {NavigationComponentProps} from 'react-native-navigation';
import {loadOneAction} from 'core/actions/movies';
import {connect} from 'react-redux';
import {itemSelector} from 'core/reselect';
import {Image, Text} from 'react-native-elements';

interface Props extends NavigationComponentProps {
  itemId: string;
  item: object;
}

const DetailsScreen: FC<Props> = (props) => {
  console.log(props);
  const {loadOne, itemId, item} = props;
  useEffect(() => {
    loadOne(itemId);
  }, []);
  return (
    <>
      {item ? (
        <View style={{alignItems: 'center'}}>
          <Image
            style={{width: 200, height: 200, resizeMode: 'contain'}}

            PlaceholderContent={<ActivityIndicator />}
            source={{uri: item.Poster}}
          />
          <Text h3>{item.Title}</Text>
          <Text h4>Year: {item.Year}</Text>
          {item.Rated && (
            <>
              <Text>Plot: {item.Plot}</Text>
              <Text>Rated: {item.Rated}</Text>
              <Text>Released: {item.Released}</Text>
              <Text>Genre: {item.Genre}</Text>
              <Text>Director: {item.Director}</Text>
              <Text>Ratings:</Text>
              {item.Ratings.map((rating) => (
                <View style={{flexDirection: 'row'}}>
                  <Text>Source: {rating.Source}</Text>
                  <Text>Value: {rating.Value}</Text>
                </View>
              ))}
            </>
          )}
        </View>
      ) : (
        <ActivityIndicator />
      )}
    </>
  );
};

const mapState = (state) => ({
  item: itemSelector(state),
});

const mapActions = (dispatch) => ({
  loadOne: (id) => dispatch(loadOneAction(id)),
});

export default connect(mapState, mapActions)(DetailsScreen);
