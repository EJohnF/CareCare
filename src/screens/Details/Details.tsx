import React, {FC, useEffect} from 'react';
import {ActivityIndicator, ScrollView, StyleSheet, View} from 'react-native';
import {NavigationComponentProps} from 'react-native-navigation';
import {loadOneAction} from 'core/actions/movies';
import {connect, ConnectedProps} from 'react-redux';
import {itemSelector} from 'core/reselect';
import {Image, Text} from 'react-native-elements';
import {RootState} from 'core/store';
import {ThunkDispatch} from 'redux-thunk';
import {Action} from 'redux';

const mapState = (state: RootState) => ({
  item: itemSelector(state),
});

const mapActions = (dispatch: ThunkDispatch<RootState, void, Action>) => ({
  loadOne: (id: string) => dispatch(loadOneAction(id)),
});

export const connector = connect(mapState, mapActions);

type PropsFromRedux = ConnectedProps<typeof connector>;

type Props = NavigationComponentProps &
  PropsFromRedux & {
    itemId: string;
  };

export const DetailsScreen: FC<Props> = ({loadOne, itemId, item}) => {
  useEffect(() => {
    loadOne(itemId);
  }, [itemId, loadOne]);
  return (
    <>
      {item ? (
        <ScrollView contentContainerStyle={styles.container}>
          <Image
            style={styles.image}
            PlaceholderContent={<ActivityIndicator />}
            source={{uri: item.Poster}}
          />
          <Text h3>{item.Title}</Text>
          <Text h4>Year: {item.Year}</Text>
          {item.Rated && (
            <>
              <Text>Plot: {item.Plot}</Text>
              <Text>Rated: {item.Rated}</Text>
              <Text>Released: {item.Released}</Text>
              <Text>Genre: {item.Genre}</Text>
              <Text>Director: {item.Director}</Text>
              <Text>Ratings:</Text>
              {item.Ratings &&
                item.Ratings.map((rating) => (
                  <View style={styles.rating} key={rating.Source}>
                    <Text>Source: {rating.Source}</Text>
                    <Text>Value: {rating.Value}</Text>
                  </View>
                ))}
            </>
          )}
        </ScrollView>
      ) : (
        <View style={styles.activityContainer}>
          <ActivityIndicator size={'large'} />
        </View>
      )}
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    marginHorizontal: 16,
  },
  activityContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  image: {
    width: 200,
    height: 200,
    resizeMode: 'contain',
  },
  rating: {
    flexDirection: 'row',
  },
});
