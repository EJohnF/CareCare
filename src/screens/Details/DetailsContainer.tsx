import {connector, DetailsScreen} from 'screens/Details/Details';

export default connector(DetailsScreen);
