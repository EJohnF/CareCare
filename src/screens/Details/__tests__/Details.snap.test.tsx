import React from 'react';
import renderer from 'react-test-renderer';
import {DetailsScreen} from '../Details';
import movie from './static/movie';

it('renders filled', () => {
  const tree = renderer
    .create(
      <DetailsScreen
        loadOne={() => {}}
        item={movie}
        itemId={movie.imdbID}
        componentId={'component1'}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders half-empty', () => {
  const tree = renderer
    .create(
      <DetailsScreen
        loadOne={() => {}}
        item={{
          Poster: movie.Poster,
          Title: movie.Title,
          Year: movie.Year,
          imdbID: movie.imdbID,
          Type: movie.Type,
        }}
        itemId={movie.imdbID}
        componentId={'component1'}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
