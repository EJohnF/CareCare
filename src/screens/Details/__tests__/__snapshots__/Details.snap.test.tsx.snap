// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders filled 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "marginHorizontal": 16,
    }
  }
>
  <View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          Object {
            "uri": "https://m.media-amazon.com/images/M/MV5BYmU1NDRjNDgtMzhiMi00NjZmLTg5NGItZDNiZjU5NTU4OTE0XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "height": 200,
            "left": 0,
            "position": "absolute",
            "resizeMode": "contain",
            "right": 0,
            "top": 0,
            "width": 200,
          }
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transition={true}
      />
      <View
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "height": 200,
              "justifyContent": "center",
              "resizeMode": "contain",
              "width": 200,
            }
          }
          testID="RNE__Image__placeholder"
        >
          <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="small"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "height": 200,
            "resizeMode": "contain",
            "width": 200,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#242424",
          "fontSize": 35,
        }
      }
    >
      Star Wars: Episode V - The Empire Strikes Back
    </Text>
    <Text
      style={
        Object {
          "color": "#242424",
          "fontSize": 27.5,
        }
      }
    >
      Year: 
      1980
    </Text>
    <Text
      style={
        Object {
          "color": "#242424",
        }
      }
    >
      Plot: 
      After the Rebels are brutally overpowered by the Empire on the ice planet Hoth, Luke Skywalker begins Jedi training with Yoda, while his friends are pursued by Darth Vader and a bounty hunter named Boba Fett all over the galaxy.
    </Text>
    <Text
      style={
        Object {
          "color": "#242424",
        }
      }
    >
      Rated: 
      PG
    </Text>
    <Text
      style={
        Object {
          "color": "#242424",
        }
      }
    >
      Released: 
      20 Jun 1980
    </Text>
    <Text
      style={
        Object {
          "color": "#242424",
        }
      }
    >
      Genre: 
      Action, Adventure, Fantasy, Sci-Fi
    </Text>
    <Text
      style={
        Object {
          "color": "#242424",
        }
      }
    >
      Director: 
      Irvin Kershner
    </Text>
    <Text
      style={
        Object {
          "color": "#242424",
        }
      }
    >
      Ratings:
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#242424",
          }
        }
      >
        Source: 
        Internet Movie Database
      </Text>
      <Text
        style={
          Object {
            "color": "#242424",
          }
        }
      >
        Value: 
        8.7/10
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#242424",
          }
        }
      >
        Source: 
        Rotten Tomatoes
      </Text>
      <Text
        style={
          Object {
            "color": "#242424",
          }
        }
      >
        Value: 
        94%
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#242424",
          }
        }
      >
        Source: 
        Metacritic
      </Text>
      <Text
        style={
          Object {
            "color": "#242424",
          }
        }
      >
        Value: 
        82/100
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders half-empty 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "marginHorizontal": 16,
    }
  }
>
  <View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          Object {
            "uri": "https://m.media-amazon.com/images/M/MV5BYmU1NDRjNDgtMzhiMi00NjZmLTg5NGItZDNiZjU5NTU4OTE0XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "height": 200,
            "left": 0,
            "position": "absolute",
            "resizeMode": "contain",
            "right": 0,
            "top": 0,
            "width": 200,
          }
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
        transition={true}
      />
      <View
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "height": 200,
              "justifyContent": "center",
              "resizeMode": "contain",
              "width": 200,
            }
          }
          testID="RNE__Image__placeholder"
        >
          <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="small"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "height": 200,
            "resizeMode": "contain",
            "width": 200,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#242424",
          "fontSize": 35,
        }
      }
    >
      Star Wars: Episode V - The Empire Strikes Back
    </Text>
    <Text
      style={
        Object {
          "color": "#242424",
          "fontSize": 27.5,
        }
      }
    >
      Year: 
      1980
    </Text>
  </View>
</RCTScrollView>
`;
